hibernate-core/src/main/java/org/hibernate/type/descriptor/java/JavaType.java

```diff
@@ -112,7 +114,7 @@ default MutabilityPlan<T> getMutabilityPlan() {
 		return ImmutableMutabilityPlan.instance();
 	}
 
-	default T getReplacement(T original, T target, SharedSessionContractImplementor session) {
+	default @NonNull T getReplacement(T original, T target, SharedSessionContractImplementor session) {
```

Comment:
I think this can be a bit stricter.

```suggestion
	default @NonNull T getReplacement(@NonNull T original, @Nullable T target, SharedSessionContractImplementor session) {
```